evaluator/evaluator/routers/dataset.py

```diff
@@ -1,22 +1,234 @@
 import logging
-from typing import List
+from typing import Dict, List
 
-from fastapi import APIRouter
+from bson import ObjectId
+from evaluate import load
+from fastapi import APIRouter, Body, Depends, FastAPI, HTTPException, Request
+from square_auth.auth import Auth
 
-from evaluator.models import DataSet
+from evaluator import mongo_client
+from evaluator.core.dataset_handler import DatasetDoesNotExistError, DatasetHandler
+from evaluator.models import Dataset
+from evaluator.mongo.mongo_client import MongoClient
+from evaluator.routers import client_credentials
+from evaluator.routers.evaluator import get_dataset_metadata
 
 logger = logging.getLogger(__name__)
-
 router = APIRouter(prefix="/dataset")
+dataset_item_type = "dataset"
+auth = Auth()
+
+
+@router.post("", status_code=201)
+async def create_dataset(
+    *,
+    dataset_name: Dataset = Body(
+        examples={
+            "Multiple-Choicesample": {
+                "summary": "Multiple-Choicesample",
+                "description": "for Multiple-Choice Dataset",
+                "value": {
+                    "name": "commensense_qa",
+                    "skill_type": "multiple-choice",
+                    "metric": "accuracy",
+                    "mapping": {
+                        "id-column": "id",
+                        "question-column": "question",
+                        "choice-columns": ["choice.text"],
+                        "choices-key-mapping-column": "choices.label",
+                        "answer-index-column": "answerkey",
+                    },
+                },
+            },
+            "Extractivesample": {
+                "summary": "Extractivesample",
+                "description": "for Extractive Dataset",
+                "value": {
+                    "name": "quoref",
+                    "skill-type": "extractive-qa",
+                    "metric": "squad",
+                    "mapping": {
+                        "id-column": "id",
+                        "question-column": "question",
+                        "context-column": "context",
+                        "answer-text-column": "answers.text",
+                    },
+                },
+            },
+        },
+    ),
+):
+    logger.debug(f"post dataset: dataset_name= {name}")
+    dataset_name_exist = mongo_client.client.evaluator.datasets.find(
+        {"dataset_name": name}
+    )
+    logger.debug(f"post anfrage")
+    if dataset_name_exist is None:
+        logger.debug(f"dataset name is None ")
+
+        if name == "quoref":
```

Comment:
Was soll das hier? Das was du hier in `dataset_mapping` schreibst wird doch im Request übergeben 🤔 Da hast du ja das `Dataset` Objekt

Reply:
wenn man die Dataset objekt direct benutzt, bekommt man folgende Fehlermeldung: document must be an instance of dict, bson.son.SON, bson.raw_bson.RawBSONDocument, or a type that inherits from collections.MutableMapping",

Dafür muss man ein dictionary definieren, um die Objekte in der Datenbank zu speichern

Reply:
Du musst Dataset (in models.py) einfach zu einen MongoModel statt einem BaseModel machen (vielleicht musst du das auch für MultipleChoiceDatasetSample und ExtractiveDatasetSample anpassen). Und dann kannst du einfach folgendes machen... : mongo_client.client.evaluator.datasets.insert_one(dataset.mongo()).

.mongo() wandelt das dann in ein dictionary um für dich.
Und beim Auslesen kannst du Dataset.from_mongo(...) benutzen.

Reply:
ich habe das Model angepasst und Änderungen gemacht